### PR TITLE
EES-4641 Use k8s errors to find out about LT not found

### PR DESCRIPTION
**Pull request by vgarvardt**
Use k8s errors to find out about LT not found - use built-in k8s errors to find out if object is not found instead of type casting errors and checking fields manually

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -37,11 +37,7 @@ jobs:
         env:
           GOOGLEAPIS_DIR: "./googleapis"
         run: |
-          go install \
-              github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway \
-              github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2 \
-              google.golang.org/protobuf/cmd/protoc-gen-go \
-              google.golang.org/grpc/cmd/protoc-gen-go-grpc
+          make protoc-plugins
```
Comment (boekkooi-fresh) [inline .github/workflows/release.yml:40]: Why not make this a requirement for `protoc` in the makefile?

File: docs/README.md [hunk 2]
```diff
@@ -93,22 +93,68 @@ kubectl apply -f charts/kagal/crd.yaml
...
+```bash
+make protoc-plugins
+```
+
+#### 3.3 Checkout required protobuf definitions
+
+In order to allow 3rd party API definitions you need to have them locally:
+
+```bash
+# go outside of the kangas source path
```
Comment (s-radyuk) [inline docs/README.md:125]:
```suggestion
# go outside of the kangal source path
```

File: Makefile [hunk 3]
```diff
@@ -41,6 +41,16 @@ apply-crd:
...
 	@kubectl apply -f charts/kangal/crd.yaml
 
+# Install required protoc plugins
+protoc-plugins:
+	@printf "$(OK_COLOR)==> Installing required protoc plugins$(NO_COLOR)\n"
+	@go install \
+	   github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway \
+	   github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2 \
+	   google.golang.org/protobuf/cmd/protoc-gen-go \
+	   google.golang.org/grpc/cmd/protoc-gen-go-grpc
```
Comment (boekkooi-fresh) [inline Makefile:51]: Do we really need to install things system wide? Can't we use something like the following to also ensure the correct version.
```
	BIN_DIR=...
	BUILD_DIR=$$(mktemp -d); \
	cd "$${BUILD_DIR}"; \
	go mod init protoc-plugins; \
	go get "github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway@v2.0.1"; \
	go build -mod=readonly -o "$${BIN_DIR}/protoc-gen-grpc-gateway" github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway; \
	rm -rf "$${BUILD_DIR}"
	"$${BIN_DIR}/protoc-gen-grpc-gateway" --version
```

File: pkg/proxy/proxy.go [hunk 4]
```diff
@@ -242,7 +241,7 @@ func (p *Proxy) Get(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		logger.Error("Could not get load test info with error:", zap.Error(err))
 
-		if k8sErr, ok := err.(k8sAPIErrors.APIStatus); ok && k8sErr.Status().Code == http.StatusNotFound {
+		if k8sAPIErrors.IsNotFound(err) {
```
Comment (boekkooi-fresh) [inline pkg/proxy/proxy.go:244]: Seems like this is unrelated to the protoc makefile changes can you split this PR please.

File: .github/workflows/release.yml (hunk 1 above)
Comment (vgarvardt) [reply .github/workflows/release.yml:40]: `make protoc` should be called for every proto file change, calling `go install ...` every time would take much more time

File: pkg/proxy/proxy.go (hunk 4 above)
Comment (vgarvardt) [reply pkg/proxy/proxy.go:244]: It is already in another dedicated commit.

File: Makefile (hunk 3 above)
Comment (vgarvardt) [reply Makefile:51]: These binaries are `protoc` plugins, installing them like this makes the whole set up much more complicated. And we expect `protoc` installed globally as well.

Comment (boekkooi-fresh) [reply Makefile:51]: Should we have protoc installed globally? Are we going to support protoc v1, v2, v3 etc?
By not providing versions we actually get a build system that will very every single time something is build.

File: .github/workflows/release.yml (hunk 1 above)
Comment (boekkooi-fresh) [reply .github/workflows/release.yml:40]: Then use make files properly and only build what is needed. For example:
```
# Protoc is used to generate code from a protobuf file
PROTOC_VERSION=3.11.4
PROTOC_ZIP=protoc-$(PROTOC_VERSION)-linux-x86_64.zip
ifeq ($(GOHOSTOS)_$(GOHOSTARCH),darwin_amd64)
    PROTOC_ZIP=protoc-$(PROTOC_VERSION)-osx-x86_64.zip
endif
PROTOC_RELEASES_URI=https://github.com/protocolbuffers/protobuf/releases/download
PROTOC_DOWNLOAD=$(PROTOC_RELEASES_URI)/v$(PROTOC_VERSION)/$(PROTOC_ZIP)

tools/protoc = $(BIN_DIR)/protoc
$(tools/protoc):
	mkdir -p "$(@D)"
	curl --fail --output "$@.zip" -L "$(PROTOC_DOWNLOAD)"
	go run tools/unzip.go "$@.zip" "$@" "bin/protoc"
	rm -f "$@.zip"
```

File: Makefile (hunk 3 above)
Comment (vgarvardt) [reply Makefile:51]: We are using latest stable `protoc` version that is `3.x` - it is pinned in the CI and in the dev instruction.

File: Makefile [hunk 5]
```diff
@@ -9,7 +9,7 @@ VERSION ?= "0.0.0-dev-$(shell git rev-parse --short HEAD)"
 # since we're using github.com/grpc-ecosystem/grpc-gateway it needs to be manually cloned and put somewhere
 # close to the project and this env var should point to "third_party/googleapis" subdir.
 # IMPORTANT - check currently used version of github.com/grpc-ecosystem/grpc-gateway to use exactly the same tag
-GOOGLEAPIS_DIR ?= "../googleapis"
+GOOGLEAPIS_DIR ?= "../grpc-gateway/third_party/googleapis"
```
Comment (diegomarangoni) [inline Makefile:12]: what do you think if we use submodules to clone third parties?
this way we ensure `googleapis` is at correct version

Comment (vgarvardt) [reply Makefile:12]: git submodules is the worst thing I've ever used for vendoring deps, I'll think how to do this in an easier way

**Comment by vgarvardt** [comment]
Ok, `protoc` stuff turned to be more than just updating README, so it goes into https://github.com/hellofresh/kangal/pull/67 to unblock small real code change as every merge to master requires rebase.

File: pkg/proxy/proxy.go [hunk 6]
```diff
@@ -242,7 +241,7 @@ func (p *Proxy) Get(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		logger.Error("Could not get load test info with error:", zap.Error(err))
```
Comment (diegomarangoni) [inline pkg/proxy/proxy.go:242]:
```suggestion
		logger.Error("Could not get load test info with error", zap.Error(err))
```
one more here

**Comment by aleksandrzhiliaev** [comment]
👍
